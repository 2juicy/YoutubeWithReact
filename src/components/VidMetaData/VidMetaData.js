import React from "react";
import "./VidMetaData.scss";
import { Button, Icon } from "semantic-ui-react";
import Rating from "../Rating/Rating";

export default function VidMetaData(props) {
  console.log(props);
  const commas = props.views
    ? props.views.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",")
    : null;

  return (
    <div className="vid-metadata">
      <h3>Video Title</h3>
      <div className="vid-stats">
        <span>{commas} views</span>
        <div className="vid-actions">
          <Rating likes={123} dislikes={12} />
          <Button basic icon labelPosition="left">
            <Icon name="share" />
            Share
          </Button>
          <Button basic icon>
            <Icon name="add circle" />
          </Button>
          <Button basic icon>
            <Icon name="ellipsis horizontal" />
          </Button>
        </div>
      </div>
    </div>
  );
}
