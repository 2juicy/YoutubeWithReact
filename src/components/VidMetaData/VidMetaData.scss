@import "../../styles/shared.scss";

.vid-metadata {
  color: #707070;
  font-size: 16px;

  h3 {
    font-weight: bold;
    color: black;
  }

  .vid-stats {
    display: flex;
    justify-content: space-between;
    width: 100%;
    align-items: center;

    .vid-actions {
      display: flex;
      align-items: center;

      & > *:not(:last-child) {
        margin-right: 8px;
      }
    }
  }
}
