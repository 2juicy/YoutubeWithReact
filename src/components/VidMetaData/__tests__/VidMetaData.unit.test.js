import React from "react";
import { shallow } from "enzyme";
import VidMetaData from "../VidMetaData";

describe("VidMetaData", () => {
  test("renders without props", () => {
    const wrapper = shallow(<VidMetaData />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with views", () => {
    const wrapper = shallow(<VidMetaData views={123145} />);
    expect(wrapper).toMatchSnapshot();
  });
});
