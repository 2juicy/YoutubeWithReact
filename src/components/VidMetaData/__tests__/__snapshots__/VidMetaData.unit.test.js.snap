// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VidMetaData renders with views 1`] = `
<div
  className="vid-metadata"
>
  <h3>
    Video Title
  </h3>
  <div
    className="vid-stats"
  >
    <span>
      123,145
       views
    </span>
    <div
      className="vid-actions"
    >
      <Rating
        dislikes={12}
        likes={123}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="add circle"
        />
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
</div>
`;
